snapcraft/_store.py

```diff
@@ -100,6 +101,119 @@ def _requires_login():
         raise
 
 
+@contextmanager
+def _requires_command_success(command, not_found_err=None, failure_err=None):
+    if isinstance(command, str):
+        cmd_list = command.split()
+    else:
+        cmd_list = command
+    try:
+        iter(cmd_list)
+    except TypeError:
+        raise
+    not_found_err = not_found_err or EnvironmentError(
+        '`{}` not found.'.format(cmd_list[0]))
+    failure_err = failure_err or EnvironmentError(
+        '`{}` failed.'.format(command))
+    try:
+        subprocess.check_call(
+            cmd_list, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    except FileNotFoundError as err:
+        raise not_found_err
+    except subprocess.CalledProcessError as err:
+        raise failure_err
+    yield
+
+
+@contextmanager
+def _requires_path_exists(path, error=None):
+    if error is None:
+        error_message = 'Required path does not exist: {}'.format(path)
+    if not os.path.exists(path):
+        raise EnvironmentError(error_message)
+    yield
+
+
+def _travis_encrypt_config(config_path):
+    """Encrypt given snapcraft config file for Travis jobs.
+
+    Depends on working `travis` CLI (boo!)
+    """
+    cmd = [
+        'travis', 'encrypt-file',
+        '--add', '--force', '--decrypt-to', '.snapcraft.cfg',
+        config_path, '.snapcraft.cfg.enc',
```

Comment:
Evan's gist puts the encripted file in a travis variable, and it puts only a small secret in the .yaml.
I don't think there's a big security difference on either approach, but it could be easier if we don't have to add another file to the repository. Do you have a reason to prefer this approach?